check-log/lib/check-log.go

```diff
@@ -230,13 +231,26 @@ func (opts *logOpts) searchLog(logFile string) (int64, int64, string, error) {
 		skipBytes = s
 	}
 
-	f, err := os.Open(logFile)
-	if err != nil {
-		return 0, 0, "", err
+	var r io.ReadSeeker
+	if opts.NoState || opts.CheckFirst {
```

Comment:
We should trace logrotate when CheckFirst option is specified, because current behavior is:

### When CheckFirst is disabled
- On first execution, just record current file size to the state file
    - At this time check-log does not check the contents of the file
- On second execution (and afterwards):
    1. Seek to previously recorded position (or go back to head if current file size is smaller than the position)
    2. Check the contents between sought position and EOF
    3. Record EOF position (aka file length) to state file

### When CheckFirst is enabled
- On first execution:
    1. Check the contents in all of the log file
    2. Record ROF position (aka file length) to state file
- On second execution (and afterwards):
    1. Seek to previously recorded position (or go back to head if current file size is smaller than the position)
    2. Check the contents between sought position and EOF
    3. Record EOF position (aka file length) to state file


As below, CheckFirst-enabled check-log behaves differently only at first execution, so it should trace logrotate same as Check-First disabled check-log. (Of course, it's apparently unnecessary when NoState enabled.)

Reply:
Thanks for your review.
I misunderstood the CheckFirst behavior, so I will accept your comment.